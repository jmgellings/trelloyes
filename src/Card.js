import React from 'react';
import './List.css';

function Card(props) {
    return (
        <section className='Card'>
            <button type='button' onClick={() => props.deleteCard(props.listIdx, props.cardIdx)}>delete</button>
            <h3>{props.title}</h3>
            <p>{props.content}</p>
        </section>
    )
}

export default Card