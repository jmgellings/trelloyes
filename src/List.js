import React from 'react';
import './List.css';
import Card from './Card.js';

function List(props) {
  return (
    <section className='List'>
      <header className='List-header'>
        <h2>{props.header}</h2>
      </header>
      <div className='List-cards'>
        {props.cards.map((card, cardIdx) => 
          <Card
            listIdx={props.listIdx}
            cardIdx={cardIdx}
            key={card.id}
            id={card.id}
            title={card.title}
            content={card.content}
            deleteCard={props.deleteCard}
          />
        )}
        <button type='button' className='List-add-button' onClick={() => props.newRandomCard(props.listIdx)}>
          + Add Random card
        </button>
      </div>
    </section>
  )
}

export default List