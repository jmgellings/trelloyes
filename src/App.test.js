import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';
import List from './List';

it('renders Card without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the Card UI as expected', () => {
  const tree=renderer
  .create(<Card name="Card" />)
  .toJSON();
expect(tree).toMatchSnapshot();
})

it('renders List without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List />, div);
  ReactDOM.unmountComponentAtNode(div);
})