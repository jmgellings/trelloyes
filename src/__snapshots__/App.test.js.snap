// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Card UI as expected 1`] = `
<section
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</section>
`;

exports[`renders the UI as expected 1`] = `
<section
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</section>
`;
